### PR TITLE
Add REST backend

### PR DESCRIPTION
**Pull request by fd0**
This is a port of the original work by @bchapuis in
https://github.com/restic/restic/pull/253

TODO:
- [x] Improve documentation

### CODE REVIEW

File: doc/REST_backend.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+an HTTP error otherwise.
+
+Response format: binary/octet-stream
+
+## POST /config
+
+Returns "200 OK" if the configuration of the request body has been saved,
+an HTTP error otherwise.
+
+## GET /{type}/
```
Comment (fawick) [inline doc/REST_backend.md]: The doc should list what would be acceptable values for {type}.

File: src/restic/backend/rest/backend_test.go [hunk 2]
```diff
@@ -0,0 +1,87 @@
+// DO NOT EDIT, AUTOMATICALLY GENERATED
```
Comment (fawick) [inline src/restic/backend/rest/backend_test.go]: Should the source and the generation recipe for the auto-generation be part of the repo?

Comment (fd0) [inline src/restic/backend/rest/backend_test.go]: Oh, it is. Here you can find the go generate comment: https://github.com/restic/restic/pull/464/files#diff-6397b881264cb4356775a0e5a3be197fR15

The file is generated from the file https://github.com/restic/restic/blob/master/src/restic/backend/test/tests.go by the program https://github.com/restic/restic/blob/master/src/restic/backend/test/generate_backend_tests.go

File: doc/REST_backend.md (hunk 1 above)
Comment (fd0) [inline doc/REST_backend.md]: Good point.

**Comment by fd0** [comment]
This works with my toy Ruby server and passes all the tests: https://gist.github.com/07c8a0cda34e257565e9

**Pull request merged**